codes/codes.go

```diff
@@ -142,3 +143,35 @@ const (
 	// DataLoss indicates unrecoverable data loss or corruption.
 	DataLoss Code = 15
 )
+
+var strToCode = map[string]Code{
+	"OK": OK,
+	"CANCELLED":/* [sic] */ Canceled,
+	"UNKNOWN":             Unknown,
+	"INVALID_ARGUMENT":    InvalidArgument,
+	"DEADLINE_EXCEEDED":   DeadlineExceeded,
+	"NOT_FOUND":           NotFound,
+	"ALREADY_EXISTS":      AlreadyExists,
+	"PERMISSION_DENIED":   PermissionDenied,
+	"RESOURCE_EXHAUSTED":  ResourceExhausted,
+	"FAILED_PRECONDITION": FailedPrecondition,
+	"ABORTED":             Aborted,
+	"OUT_OF_RANGE":        OutOfRange,
+	"UNIMPLEMENTED":       Unimplemented,
+	"INTERNAL":            Internal,
+	"UNAVAILABLE":         Unavailable,
+	"DATA_LOSS":           DataLoss,
+	"UNAUTHENTICATED":     Unauthenticated,
+}
+
+// UnmarshalJSON unmarshals b into the Code.
+func (c *Code) UnmarshalJSON(b []byte) error {
```

Comment:
From Unmarshaler's godoc:
`By convention, to approximate the behavior of Unmarshal itself, Unmarshalers implement UnmarshalJSON([]byte("null")) as a no-op.`
https://golang.org/pkg/encoding/json/#Unmarshaler